File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.java
class Solution {
    public int findMin(int[] nums) {
        int l = 0;
        int h = nums.length - 1;
        int min = Integer.MAX_VALUE;
        while (l <= h)
        {
            int mid = (h - l)/2 + l;
            min = Math.min(min, nums[mid]);
            if (nums[mid] < nums[h])
            {
                h = mid - 1;
            }
            else
            {
                l = mid + 1;
            }
        }
        return min;
    }
}

File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.py
class Solution(object):
    def findMin(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        l = 0
        h = len(nums) - 1
        res = nums[l]
        while l <= h:
            if nums[l] <= nums[h]:
                res = min(res, nums[l])
                break
            mid = (l + h) // 2
            res = min(res, nums[mid])
            if nums[l] <= nums[mid]:
                l = mid + 1
            else:
                h = mid - 1
        return res

            